### DISCUSSION TITLE
Query about compatibility with zsh-history-substring-search, zsh-completions and zsh-autosuggestions

### DISCUSSION THREAD

**Discussion by blueray453** [marlonrichert/zsh-autocomplete#456] [Q&A] [answered]
Do I need to install zsh-users/zsh-completions or zsh-autocomplete take care of it?

Can I use zsh-users/zsh-autosuggestions with zsh-autocomplete? Then how does it work?

I assume i do not need to use  zsh-users/zsh-history-substring-search. How does it work? What is the workflow (shortcut) of zsh-autocomplete to use zsh-users/zsh-history-substring-search like feature?

**Comment by marlonrichert** [comment, marked as answer]
> Do I need to install zsh-users/zsh-completions or zsh-autocomplete take care of it?

Take care of what?

> Can I use zsh-users/zsh-autosuggestions with zsh-autocomplete? 

Yes. 

> Then how does it work?

The same. Zsh Autocomplete does not override [zsh-users/zsh-autosuggestions](https:/github.com/zsh-users/zsh-autosuggestions)’ keybindings.

> I assume i do not need to use zsh-users/zsh-history-substring-search.

That is correct.

> How does it work? What is the workflow (shortcut) of zsh-autocomplete to use zsh-users/zsh-history-substring-search like feature?

It doesn’t use it. Instead, when you press ⬆️, Zsh Autocomplete opens a menu with recent command line history for you.

**Reply by blueray453** [reply]
> Take care of what?

I mean does it download zsh-completions and use it. Or, do i need to download it separately.

**Reply by marlonrichert** [reply]
No, you need to install those separately.
